### ISSUE TITLE
bug(tiktok): cannot follow users - app immediately unfollows them

### ISSUE THREAD

**Issue by mrmicael** [ReVanced/revanced-patches#2098]
### Type

Other

### Bug description

When I follow someone, it automatically unfollows them. I've already reported this on the Discord server and two other people have reported the same issue. I've followed all steps and tried in multiple apps and devices.

### Steps to reproduce

1 . Run Tiktok with patches
2 . log into your account
3 . follow someone

### Relevant log output

```shell
There's nothing i can put here.
```

**Comment by CrazyWolf13** [comment]
Have we got any news here?
Still experiencing this issue.

**Comment by Suikaino** [comment]
> Maybe by blocking tiktok analytics servers will work to bypass client has been tempered or not.

Do you have their analytics server lists?

**Comment by Suikaino** [comment]
> > > Maybe by blocking tiktok analytics servers will work to bypass client has been tempered or not.
> > 
> > 
> > Do you have their analytics server lists?
> 
> Try this `https://github.com/hagezi/dns-blocklists?tab=readme-ov-file#calling-native-tracker---broadband-tracker-of-devices-services-and-operating-systems-`

Holy moly, it works.
Thank you

**Comment by CrazyWolf13** [comment]
@Suikaino  @cyberboh Thanks for the idea, just tested with the adfilter enabled through pihole, however I can still not follow anyone, but on my end, (most likely) tiktok shadowbanned my account, as I can not follow on any client -> ios, android, windows, web, revanced etc.

**Comment by CrazyWolf13** [comment]
@oSumAtrIX  @LisoUseInAIKyrios Yes this is still an issue, I just tested with the adlists filter enabled, on a new device + network, and I couldn't get it to work.

Also it seems to appear on new accounts.

Is there anything we can provide to get this fixed? Any logs or testing?
I'd really love to see this fixed, as this is a ground-function from tiktok that is not working, and this issues seems to be open for 2 years now.

**Comment by CrazyWolf13** [comment]
Small update, have just received an answer from tiktok, which is kinda promising, seems like the began heavily restricting the older tiktok versions. After switching to the latest version I did not experience anymore issues.

Maybe the tiktok version bump requested here: #3282 will help and even resolve this issue.

FYI: @oSumAtrIX

**Comment by CrazyWolf13** [comment]
Issue persists with latest 36.5.4 update.

**Comment by panagiotis94** [comment]
It still persists

**Comment by CrazyWolf13** [comment]
Hi 
Forgot to update here, but the last version bump resolved the issue for me for quite some time, now it seems to happen again, could we get another bump?

**Comment by CrazyWolf13** [comment]
@soliviantar Yeah my testing showed, that it seems like tiktok shadownbans revanced users(all sessions cannot really follow/like, though comments seem to go through after a longer hold-back delay), I once deleted the tiktok app for a period (some weeks but cannot be exact here, as I don't know anymore) and then re-patched with the version bump, after that it worked again for some months, now it's back.

**Comment by CrazyWolf13** [comment]
@Be2iz this is known, we are "in desperate need" of a version bump due to having such an outdated version tiktok shadowbanns /locks the majority of revanced users and limits interaction and fully blocks following etc.

Check this out: https://github.com/ReVanced/revanced-patches/pull/4920

**Comment by CrazyWolf13** [comment]
Linking to https://github.com/ReVanced/revanced-patches/issues/4221 as the root cause, because updating to the latest app version immediately fixes the "cannot follow users" issue.

**Comment by CrazyWolf13** [comment]
@LisoUseInAIKyrios I guess we don't know for sure, but last time a version bump was done it was exactly the same thing.
Tiktok "Shadowbanned" me after updating the app it worked for a couple of months I guess till the version got too old and then I was shadowbanned again. Changing to an updated version always fixed the issue.

I couldn't test with the latest version as they fail to build. But I'll happily try to find the initial Commits by @Nuckyz and try to manually apply them.

I also hammered the support a bit trying to get some info and one of my issue requests I think made it to a higher level where they kinda admitted that they "slow down /limit" older app versions
